pilot/pkg/serviceregistry/federation/controller.go

```diff
@@ -920,9 +923,23 @@ func (c *Controller) deleteService(service *federationmodel.ServiceMessage, exis
 }
 
 func (c *Controller) addServiceToStore(service *model.Service, instances []*model.ServiceInstance) {
+	// Check existing services in the serviceStore, those can be created before importing services from a federation peer.
+	eventType := model.EventAdd
+	found := false
+	for i, s := range c.serviceStore {
+		if s.Hostname == service.Hostname {
+			c.serviceStore[i] = service
+			found = true
+			break
+		}
+	}
+	if found {
```

Comment:
I think this logic should just be added to the loop, before the break, removing the need for `found`, as it's only used once and the loop exits once something is found.

Reply:
This change has been updated in the PR . I added comments about proxy config_dump in OSSM-1211.
This change should fix the locality setting in an imported service endpoint_config.

However, failover is still not working after this change. After import a service as local, bookinfo reviews always reach a remote ratings endpoint instead of the existing local one. and cannot failover after scale the rating pod.
WIP

Reply:
[https://github.com/maistra/istio/blob/maistra-2.1/pilot/pkg/xds/eds.go#L166](https://github.com/maistra/istio/blob/maistra-2.1/pilot/pkg/xds/eds.go#L166)
The EDSCacheUpdate method overwrites endpoints SA in line 166 
That explains why we don't have the existing local service default SA after importing a service as local.

I added a change in this PR federation controller line939 and will test that change in two clusters tomorrow
https://github.com/maistra/istio/pull/532/files#diff-424decff1e2dc8267426e1f8160e8c40f628ff1d88bdd9885564294fd72197aeR939

@rcernich  Do I need to add a similar change in the `updateServiceInStore` method ?